### ISSUE TITLE
Run the standalone form feed with the current user

### ISSUE THREAD

**Issue by oriolroger** [backdrop-contrib/feeds#138]
One of the major changes in the latest Drupal 7 versions was that the imports are ran as the user 1, when using the standalone import form ([see here](https://www.drupal.org/project/feeds/releases/7.x-2.0-beta5)).

We woder if there is any option to disable this, and run the import with the current user, like it was before? We're experiencing some problems, because we need the current user to store some data on the database, during the parse of the feed.

Thanks!

**Comment by argiepiano** [comment]
I may not be understanding your question, but can't you actually tell Feeds what Author to use when create the nodes (or other entities)? This is done in the Processor step in the UI

<img width="628" alt="Screen Shot 2023-07-20 at 10 09 50 AM" src="https://github.com/backdrop-contrib/feeds/assets/9938978/e41dc9c6-bd51-4c50-ac52-fa1e65af0ee0">

EDIT: OK, reread your question, and read the attached article - I understand your question now and see that this was changed in D7 as well. I think it'd be good to allow that option to revert this change.

**Comment by argiepiano** [comment]
OK, reread your question. So, you want Feeds to create a node and use the uid of the current user?

**Comment by oriolroger** [comment]
Sorry, that's my fault, the explaining was not very accurate. This has to be with the new FeedsAccountSwitcher class, that as far as I understand, it changes the current user to "root" (uid = 1), while the whole import process is running in the background. I see this in _function switchTo($account)_, in the FeedsAccountSwitcher class.

I need to avoid this, because I have to store the current user uid in the database while the importing is running. Currently I'm patching the Feeds module, commenting the line _$this->switchAccount()_ in FeedsSource.inc:

```
  protected function acquireLock() {
    if (!lock_acquire("feeds_source_{$this->id}_{$this->feed_nid}", 60.0)) {
      throw new FeedsLockException(t('Cannot acquire lock for source @id / @feed_nid.', array('@id' => $this->id, '@feed_nid' => $this->feed_nid)));
    }
    // Switch account.
    //$this->switchAccount();
  }
```

I'd like to know if there's an option to do this, without having to patch the module.

Many thanks again!

**Comment by argiepiano** [comment]
Yes, I see that now. From what I can tell, that switch is hard-coded, so there is no current way to avoid the switch. But it looks like it may be possible to add a configuration option to the UI to skip this change - that shouldn't be too difficult, I think.

**Comment by MegaChriz** [comment]
See also this issue on the Feeds issue queue on drupal.org: https://www.drupal.org/project/feeds/issues/3375914

I think that there is a workaround possible.

**Comment by argiepiano** [comment]
Yes, that workaround should be possible, but it would affect all importers in your site. The D7 variable `feeds_source_class` is now stored in `feeds.settings.json`

**Comment by MegaChriz** [comment]
Well, you could conditionally override `$this->accountSwitcher`. 🙂 You have access to the importer ID in the constructor of classes extending FeedsSource.

**Comment by oriolroger** [comment]
Thank you both for your answers. I'll take a look and try to add a UI config option to disable the user switch.

**Comment by irinaz** [comment]
@oriolroger , are you still looking into this issue or you need help here?

**Comment by oriolroger** [comment]
Hi @irinaz, I fixed this issue adding the attached patch.

Thanks.

[feeds-Avoid-switch-user-to-root.patch](https://github.com/user-attachments/files/24823940/feeds-Avoid-switch-user-to-root.patch)

**Comment by irinaz** [comment]
@oriolroger , do you want to submit pull request or you prefer that someone else deal with it?

**Comment by oriolroger** [comment]
I didn't fully test it because it was urgent, it works for our platform. So if someone wants to take it from here and submit a pull request, that would be better.

Thanks.
